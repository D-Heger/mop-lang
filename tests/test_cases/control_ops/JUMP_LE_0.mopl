; Basic JUMP.LE.0 test - should jump when top of stack is less than or equal to 0
PUSH 0
JUMP.LE.0 ZERO_OR_NEGATIVE_LABEL
PRINT "Should not print this"
HALT

ZERO_OR_NEGATIVE_LABEL:
PRINT "Jumped successfully"

; Test with a negative number
PUSH -1
JUMP.LE.0 ANOTHER_ZERO_OR_NEGATIVE_LABEL
PRINT "Should not print this either"
HALT

ANOTHER_ZERO_OR_NEGATIVE_LABEL:
PRINT "Jumped successfully again"
HALT