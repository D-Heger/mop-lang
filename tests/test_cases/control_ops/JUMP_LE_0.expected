Jumped successfully
Jumped successfully again